diff --git a/record.c b/record.c
index a7bfc31..349241d 100644
--- a/record.c
+++ b/record.c
@@ -1325,7 +1325,7 @@ PRESERVE_NONE gc_obj record(bc instr, bc *pc, gc_obj *stack, vm_state *state,
     {
       auto t = record_current_trace(state);
       auto s = to_symbol(t->consts[c.loc]);
-      if (s->opt >= 0) {
+      if (false) {
         s->opt = 1;
         v1 = add_const(state, s->val);
       } else {
