diff --git a/record.c b/record.c
index 4afdf00..e67b17b 100644
--- a/record.c
+++ b/record.c
@@ -111,9 +111,7 @@ static void clear_trace_state(trace_state *ts) {
   arrfree(ts->debug_ops);
   *ts = (trace_state){};
 }
-static void free_snap(snap *snap) {
-  arrfree(snap->side_exit_jcc_locs);
-}
+static void free_snap(snap *snap) { arrfree(snap->side_exit_jcc_locs); }
 static void free_trace(trace *trace) {
   arr_for_each(trace->snaps, snap) { free_snap(&snap); }
   arrfree(trace->ins);
@@ -1322,7 +1320,7 @@ PRESERVE_NONE gc_obj record(bc instr, bc *pc, gc_obj *stack, vm_state *state,
     {
       auto t = record_current_trace(state);
       auto s = to_symbol(t->consts[c.loc]);
-      if (s->opt >= 0) {
+      if (false) {
         s->opt = 1;
         v1 = add_const(state, s->val);
       } else {
