diff --git a/record.c b/record.c
index 8b8299d..86a8459 100644
--- a/record.c
+++ b/record.c
@@ -1466,7 +1466,7 @@ PRESERVE_NONE gc_obj record(bc instr, bc *pc, gc_obj *stack, vm_state *state,
       assert(is_closure(stack[instr.v1]));
       assert(is_func(code));
       auto func = to_func(code);
-      if (func->poly_cnt < 4) {
+      if (false) {
         func->poly_cnt |= 1;
         auto clo_c = add_const(state, stack[instr.v1]);
         add_inst(state, IR(.op = IR_EQ, .op1 = clo, .op2 = clo_c));
